pkg/api/addons_test.go

```diff
@@ -3834,17 +3992,22 @@ func concatenateDefaultAddons(addons []KubernetesAddon, version string) []Kubern
 }
 
 func overwriteDefaultAddons(addons []KubernetesAddon, version, kubernetesImageBase, kubernetesImageBaseType string) []KubernetesAddon {
+	overrideAddons := make(map[string]KubernetesAddon)
+	for _, addonOverride := range addons {
+		overrideAddons[addonOverride.Name] = addonOverride
+	}
+
 	var ret []KubernetesAddon
 	defaults := getDefaultAddons(version, kubernetesImageBase, kubernetesImageBaseType)
-	for _, addonOverride := range addons {
```

Comment:
@jackfrancis With this logic, if `overwriteDefaultAddons` is called with more than 1 overwrite addon, it would result in duplicate values  for `overwriteAddons[1:]`. So I've updated the logic to just do a single pass of the array.

Reply:
yikes, thanks!